### PR TITLE
feat(src): Implemented Dark Mode on Scan

### LINKED ISSUE #1101: Dark Mode 🌔
#### What would you like to be added:
A site wide dark mode which defaults to system/browser settings and can be toggled on/off.

#### Why is this needed:
- It is easier on the eyes
- It looks good
- All the cool kids have it 😉

### PR DESCRIPTION
**Pull request by siradji**
#### Which issue(s) does this PR fixes?:

Fixes #1101 

#### Sample Links & Screenshots:

Link: 
<details>
<summary>Desktop Screenshot</summary>

</details>
<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on multiple web browsers
- [ ] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [ ] No console errors
- [ ] Unit tests*
- [ ] Added e2e tests*

### CODE REVIEW

**Review by nichellekoh — CHANGES_REQUESTED** [review]

**Comment by siradji** [comment]
> Dark Mode configurations is not implemented on Individual and Index Auction pages Screenshot 2022-05-25 at 9 48 11 AM
> 
> Screenshot 2022-05-25 at 9 49 15 AM

yeah , i was aware of that. No auctions was available then.

**Comment by joeldavidw** [comment]
@siradji Let's use the `bg-gray-50` as the default bg for text in the following:
- Table cells (Use the current darker bg for on hover)
- Alert boxes

image

image

**Comment by siradji** [comment]
@joeldavidw, can you clarify a bit more on the

> @siradji Let's use the `bg-gray-50` as the default bg for text in the following:
> 
>     * Table cells (Use the current darker bg for on hover)
> 
>     * Alert boxes
> 
> 
> image
> 
> image

@joeldavidw, can you clarify a bit more on the

**Comment by joeldavidw** [comment]
> @joeldavidw, can you clarify a bit more on the
> 
> > @siradji Let's use the `bg-gray-50` as the default bg for text in the following:
> > ```
> > * Table cells (Use the current darker bg for on hover)
> > 
> > * Alert boxes
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > image
> > image
> 
> @joeldavidw, can you clarify a bit more on this

Use the lighter gray as the default and the darker one when its being hovered. Basically, use the lighter gray (in the picture) as the background to help with text readability.

**Comment by siradji** [comment]
@joeldavidw, @nichellekoh

File: src/pages/vaults/_components/commons/VaultHealthBar.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export function VaultHealthBar (props: VaultHealthBarProps): JSX.Element {
           />
         </div>
       </div>
-      <div className='mt-0.5 w-full flex flex-wrap text-sm text-gray-500'>
+      <div className='mt-0.5 w-full flex flex-wrap text-sm text-gray-500 text-gray-100'>
```
Comment (kyleleow) [inline src/pages/vaults/_components/commons/VaultHealthBar.tsx:39]:
```suggestion
      <div className='mt-0.5 w-full flex flex-wrap text-sm text-gray-500 dark:text-gray-100'>
```

File: tailwind.config.js [hunk 2]
```diff
@@ -47,36 +60,89 @@ module.exports = {
...
         50: '#F7FBFE',
         100: '#EFF6FD',
         200: '#D6EAF9',
+        300: '#BDDDF5',
         500: '#5AA9E6',
-      }
+        600: '#5198CF',
+        700: '#3A6C92'
+      },
+      blueDark: {
```
Comment (thedoublejay) [inline tailwind.config.js:137]: let's not use camel case on CSS classes. darkblue or dark-blue is better.

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: tailwind.config.js [hunk 3]
```diff
@@ -23,6 +24,18 @@ module.exports = {
         800: '#85005B',
         900: '#5C003F'
       },
+      primaryDark: {
+        25: '#2F0421',
```
Comment (thedoublejay) [inline tailwind.config.js:28]: @kyleleow @siradji can we align these with LW?

File: src/layouts/Default.tsx [hunk 4]
```diff
@@ -42,23 +42,25 @@ export function Default (props: PropsWithChildren<ScanAppProps>): JSX.Element |
...
-                    {props.children}
-                  </main>
-                  <Footer />
-                </PoolPairsProvider>
-              </SupplyProvider>
-            </StatsProvider>
-          </StoreProvider>
-        </WhaleProvider>
-      </NetworkProvider>
+      <ThemeProvider theme='light'>
```
Comment (thedoublejay) [inline src/layouts/Default.tsx:45]: Change this to whatever is on the local storage

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: src/layouts/Default.tsx [hunk 5]
```diff
@@ -43,37 +43,39 @@ export function Default (props: PropsWithChildren<ScanAppProps>): JSX.Element |
...
+        <NetworkProvider>
```
Comment (thedoublejay) [inline src/layouts/Default.tsx:47]: Another thing is, do we want this to be the first layer of provider? Not sure if the other providers depend on this, could be moved to the inner part

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: tailwind.config.js (hunk 3 above)
Comment (kyleleow) [reply tailwind.config.js:28]: Scan is actually following figma color system, LW should be the one updating
image

File: tailwind.config.js (hunk 2 above)
Comment (kyleleow) [reply tailwind.config.js]: updated

File: src/layouts/Default.tsx (hunk 4 above)
Comment (kyleleow) [reply src/layouts/Default.tsx]: updated

File: src/layouts/Default.tsx (hunk 5 above)
Comment (kyleleow) [reply src/layouts/Default.tsx]: updated

File: tailwind.config.js [hunk 6]
```diff
@@ -47,36 +60,89 @@ module.exports = {
...
         100: '#EFF6FD',
         200: '#D6EAF9',
+        300: '#BDDDF5',
         500: '#5AA9E6',
-      }
+        600: '#5198CF',
+        700: '#3A6C92'
+      },
+      'dark-blue': {
+        50: 'rgba(76, 184, 255, 0.15)',
```
Comment (thedoublejay) [inline tailwind.config.js:138]: We should use hex here right? Because it uses opacity also on tailwind

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: tailwind.config.js (hunk 6 above)
Comment (kyleleow) [reply tailwind.config.js]: updated

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**